clang/test/SemaTemplate/ms-function-specialization-class-scope.cpp

```diff
@@ -156,7 +156,13 @@ namespace UsesThis {
     auto h<int>() -> decltype(this); // expected-error {{'this' cannot be used in a static member function declaration}}
   };
 
-  template struct A<int>; // expected-note 3{{in instantiation of}}
+  template struct A<int>; // expected-note {{in instantiation of}}
+  template<> template<> void A<int>::f<int>();
+  template<> template<> void A<int>::g<int>();
+  void test1() {
+    A<int>().f<int>(); // expected-note {{in instantiation of}}
+    A<int>().g<int>(); // expected-note {{in instantiation of}}
+  }
```

Comment:
There weren't test failures here, right?
(Changes here are hopefully not hiding the failures)

Reply:
No, we are still producing the same bunch of diagnostics, except now only when we use the function, instead of when instantiating the enclosing class.
This matches the behavior of MSVC.